client/elements/text/sc-text-image.html

```diff
@@ -83,6 +91,15 @@
           }
 
           _handleResponse() {
+              if (!this.imageData || !this.imageData[0]) {
+                return;
+              };
+              let pageNumber = 0;
+              for (let item = 0; item < this.imageData.length; item++) {
+                  if (this.imageData[item].pageNumber === this.pageNumber) {
+                      pageNumber = item;
+                  }
```

Comment:
If we don't put a `break` statement here, this loop will keep going even after it's found the element we're looking for.

Reply:
I see. But I replaced the whole thing with your loop now. But thanks for the tip for next time i.e. as soon as value is found, break.